activerecord/test/cases/migration/foreign_key_test.rb

```diff
@@ -60,6 +60,77 @@ def test_add_foreign_key_inferes_column
         assert_equal("fk_rails_78146ddd2e", fk.name)
       end
 
+      class CreateCitiesAndHousesWithoutColumnMigration < ActiveRecord::Migration
+        def change
+          create_table("cities") { |t| }
+
+          create_table("houses") do |t|
+            t.column :city_id, :integer
+          end
+        end
+      end
+
+      class AddForeignKeyMigration < ActiveRecord::Migration
+        def change
+          add_foreign_key :houses, :cities
+        end
+      end
+
+      def test_remove_foreign_key_with_prefix_and_suffix
+        ActiveRecord::Base.table_name_prefix = 'prefix_'
+        ActiveRecord::Base.table_name_suffix = '_suffix'
+
+        migration = CreateCitiesAndHousesWithoutColumnMigration.new
+        add_key_migration = AddForeignKeyMigration.new
+
+        silence_stream($stdout) do
+          migration.migrate(:up)
+          add_key_migration.migrate(:up)
+        end
+ 
+        assert_equal 1, @connection.foreign_keys('prefix_houses_suffix').size 
+
+        silence_stream($stdout) { add_key_migration.migrate(:down) }
+
+        assert_equal [], @connection.foreign_keys('prefix_houses_suffix')
+      ensure
+        silence_stream($stdout) { migration.migrate(:down) } 
+        ActiveRecord::Base.table_name_prefix = ''
+        ActiveRecord::Base.table_name_suffix = ''
+      end
+
+      def test_add_foreign_key_with_prefix_and_suffix
```

Comment:
`test_remove_foreign_key_with_prefix_and_suffix` and `test_add_foreign_key_with_prefix_and_suffix` are very similar. It's probably worth combining them. Then we can add and remove the foreign key in a single test. Something along the lines of

`test_managing_foreign_keys_with_prefix_and_suffix`